csharp/src/Google.Protobuf/UnsafeCollectionOperations.cs

```diff
@@ -0,0 +1,118 @@
+#region Copyright notice and license
+
+// Protocol Buffers - Google's data interchange format
+// Copyright 2008 Google Inc.  All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file or at
+// https://developers.google.com/open-source/licenses/bsd
+
+#endregion
+
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Security;
+using Google.Protobuf.Collections;
+
+namespace Google.Protobuf
+{
+    /// <summary>
+    /// Provides unsafe operations for working with <see cref="RepeatedField{T}"/> instances. These
+    /// methods are referred to as "unsafe" due to the fact that they potentially expose the backing
+    /// buffer of a <see cref="RepeatedField{T}"/> to the application.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// The methods in this class should only be called if it is guaranteed that the
+    /// <see cref="RepeatedField{T}"/> will not be modified while the returned <see cref="Span{T}"/>
+    /// is in use. Modifying a <see cref="RepeatedField{T}"/> while a <see cref="Span{T}"/> is in
+    /// use can lead to unexpected and undesirable consequences in your application, and will likely
+    /// be difficult to debug. Proceed with caution!
+    /// </para>
+    /// <para>
+    /// Modifying a <see cref="RepeatedField{T}"/> while a <see cref="Span{T}"/> is in use can have
+    /// a number of significant side effects:
+    /// </para>
+    /// <list type="bullet">
+    /// <item>
+    /// <description>
+    /// The <see cref="Span{T}"/> may or may not reflect the changes made to the
+    /// <see cref="RepeatedField{T}"/>.
+    /// </description>
+    /// </item>
+    /// <item>
+    /// <description>
+    /// If the backing array of the <see cref="RepeatedField{T}"/> is reallocated, further updates
+    /// to the <see cref="Span{T}"/> will not be visible.
+    /// </description>
+    /// </item>
+    /// <item>
+    /// <description>
+    /// The state of the <see cref="Span{T}"/> is undefined after any modifications to the <see
+    /// cref="RepeatedField{T}"/>.
+    /// </description>
+    /// </item>
+    /// </list>
+    /// </remarks>
+    [SecuritySafeCritical]
+    public static class UnsafeCollectionOperations
+    {
+        /// <summary>
+        /// Returns a <see cref="Span{T}"/> that wraps the backing array of the specified
```

Comment:
The comment should mention:

Values in the `Span<T>` shouldn't be set to `null`. Items should be removed from a RepeatedField with Remove or RemoveAt.